### ISSUE TITLE
Fusion of optimizer step into the backward pass (FSDP / FSDP2)

### ISSUE THREAD

**Issue by vadimkantorov** [volcengine/verl#2558]
For further peak VRAM reduction, is the technique of `tensor.register_post_accumulate_grad_hook(...)` from https://docs.pytorch.org/tutorials/intermediate/optimizer_step_in_backward_tutorial.html applicable in Verl / FSDP?

This should enable deallocating gradients much earlier and reducing needed peak VRAM.

Although, this should not be compatible with micro-batching / gradient accumulation, right?

**Comment by vermouth1992** [comment]
It is not in general applicable as fusing optimizer step into backward pass requires that there is only one micro batch (e.g., no gradient accumulation), which is not the case in RL training
